dm/dm/master/openapi_test.go

```diff
@@ -151,7 +155,7 @@ func (t *openAPISuite) TestOpenAPIWillNotStartInDefaultConfig(c *check.C) {
 	}), check.IsTrue)
 	c.Assert(s1.openapiHandles, check.IsNil)
 	defer s1.Close()
-	cancel()
+	defer cancel() // this cancel must call before s.Close() to avoid deadlock
```

Comment:
`cancel()` is the last call, defer or not makes no difference

Reply:
when some `c.Assert` failed， the last line cancel will not be called 😂  see this [comment](https://github.com/pingcap/tiflow/pull/4556#pullrequestreview-878430557)

Reply:
then `defer` should be moved before `assert`, else it won't be called either.

Reply:
sorry I didn't make myself clear

when some `c.Assert` that before `s.Close` and last line's `cancel()` failed, this test will blocked forever in the first `defer s.Close()` because the ctx is not cancelled

so `defer cancel` just needs  to be called before `s.Close`.

Reply:
> when some c.Assert that before s.Close and last line's cancel() failed, this test will blocked forever in the first defer s.Close()

defer runs when code runs past that point, in this case `defer s.Close()`  which is after the failed `c.Assert` will never run, see:

```go
func Test(t *testing.T) {
	require.True(t, false)
	defer fmt.Println("never run")
}
```

Reply:
yes, but test failed at L126 `c.Assert(result.Code(), check.Equals, http.StatusOK)`, new added defer is before this line

what i want to change is make sure `defer cancel() ` called before `defer s2.Close()`